src/configure.ts

```diff
@@ -25,42 +25,35 @@ export function configureGqlx<TApi, TData>(options: GatewayOptions<TApi, TData>)
   const schema = createSchema(services);
   const gqlxServer: GraphQLServer<TApi, TData> = {
     applyMiddleware(app) {
-      const root = paths.root || defaultRootPath;
+      const server = new ApolloServer({
+        tracing,
+        cacheControl,
+        uploads: {
+          maxFiles,
+          maxFileSize,
+        },
+        schema: schema.get(),

```

Comment:
This is the problem. Its static, but it must be dynamic.

Reason: We update the schema on the fly (e.g., in the gateway) and these updates must be applied *without* restarting the server.

Reply:
This seems to be tricky with the new version, but I'm looking into the source code to identify any possibilities for that. Also found several similar issues https://github.com/apollographql/apollo-server/issues/1275, https://github.com/apollographql/apollo-server/issues/2481.

Reply:
Just pushed changes [`d387ea0` (#10)](https://github.com/graphql-extended/gqlx-apollo-express-server/pull/10/commits/d387ea0af159150d2caa3685dcb1ac348cc2adbd), which allows schema reload in the runtime. I have not tested yet all use-cases, but the hot-reload itself works as expected. Enabled `gateway` with the custom implementation for apollo server does the thing properly.

Reply:
There is also a test repository (still locally), which I use for testing. I will publish it soon in order to ensure no BC.